matminer/featurizers/bandstructure.py

```diff
@@ -108,15 +109,28 @@ class BandFeaturizer(BaseFeaturizer):
     Featurizes a pymatgen band structure object.
     """
 
-    def __init__(self):
-        pass
+    def __init__(self, kpoints=None, atol=0.01, include_symeqks=False,
+                 weight=1):
+        self.atol = atol
+        self.include_symeqks = include_symeqks
+        self.kpoints = kpoints
+        self.weight = weight
 
     def featurize(self, bs):
         """
         Args:
             bs (pymatgen BandStructure or BandStructureSymmLine or their dict):
                 The band structure to featurize. To obtain all features, bs
                 should include the structure attribute.
+            kpoints ([1x3 numpy array]): list of fractional coordinates of
+                k-points at which energy is extracted.
+            atol (float): the absolute tolerance for fractional coordinate
+                below which two coordinates are considered the same
+            include_symeqks (bool): if True, when looking for energy of the
```

Comment:
it is unclear to me from the documentation what is the purpose of setting this to True. Do I get different results? If so, for which features?

Reply:
This is for when a given kpoint is not available in the input BandStructure object but its symmetrically equivalent one is; so instead of returning nan we find that symmetrically equivalent kpoint and return its energy. I thought this is what you wanted? It will be more useful if the input BandStructure is in uniform kmesh.

Reply:
Btw, if the purpose of this is to automatically automatically add more k-points if there are symmetrically equivalent kpoints to the ones the user provides (e.g., the user provides [0,0,0.5] and the code automatically adds [0.5, 0, 0] and [0, 0.5, 0] so you get features like n_0.0;0.0;0.5_en, n_0.0;0.5;0.0_en:, n_0.5;0.0;0.0_en:), I would suggest to remove this parameter. Typically one wants to know how many features they are getting out in advance so that the number of features will match for different materials.  Instead, I could see there being a use for a separate featurize where you provide a list of user k-points and it returns a list of weights for those kpoints.

Reply:
no it does not provide additional unwanted features it just looks among all symmetrically equivalent k-points to find the energy of the given k-point.

Reply:
Oh I see, the documentation doesn't say that at all. In that case, there should be a parameter called ``use_closest_k`` and for which the documentation says:

"If True, will substitute data from the closest available k-point if a k-point specified by the user is not present in the band structure".

Is this the correct description? If so, not sure why the recommendation is for that to be False.

* If all the specified k-points are already there in the band structure, then this parameter should not affect speed if implemented properly.
* If all the specified kpoints are not already there in the band structure, then this parameter becomes necessary to produce data